### PR TITLE
Refactor bodyToHtml, and allow onHighlightClicked

### PR DESCRIPTION
**Pull request by richvdh**
In the spirit of keeping PRs tractable, here is a minor one which mostly just
refactors bodyToHtml (it's not perfect, but it's less terrible), and adds
'onHighlightClick'.

### CODE REVIEW

File: src/HtmlUtils.js [hunk 1]
```diff
@@ -61,77 +67,88 @@ module.exports = {
...
+     * highlight: true to highlight as a search match
+     */
+    _createSpan(spanBody, highlight) {
+        var spanProps = {
+            key: this._key++,
+        };
+
+        if (highlight) {
+            spanProps.onClick = this.onHighlightClick;
+            spanProps.className = "mx_MessageTile_searchHighlight";
```
Comment (Kegsay) [inline src/HtmlUtils.js]: It's sad that a generic `Highlighter` class in a generic `HtmlUtils` class is specifying such a precise `className` here.

File: src/HtmlUtils.js [hunk 2]
```diff
@@ -61,77 +67,88 @@ module.exports = {
...
-            if (content.format === "org.matrix.custom.html") {
-                var safeBody = sanitizeHtml(content.formatted_body, sanitizeHtmlParams);
-                bodyList = this._applyHighlights(safeBody, highlights, true, k);
-            }
-            else {
-                bodyList = this._applyHighlights(originalBody, highlights, true, k);
-            }
-            body = bodyList;
+module.exports = {
+    bodyToHtml: function(content, highlights, opts) {
```
Comment (Kegsay) [inline src/HtmlUtils.js]: Docs on args would be nice. I have no idea what `highlights` are, nor what possible options are present in `opts`. I can guess `content` is the event content based on `:132`.

**Comment by Kegsay** [comment]
It's less bad. I don't like the juggling of unsafe HTML snippets, but at least we use `safe` as a prefix.

File: src/HtmlUtils.js (hunk 1 above)
Comment (richvdh) [inline src/HtmlUtils.js]: yes it is. factored out.

File: src/HtmlUtils.js (hunk 2 above)
Comment (richvdh) [inline src/HtmlUtils.js]: I added some words.

**Comment by richvdh** [comment]
PTAL?

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**